packages/core/src/transactions/utils/transactions.ts

```diff
@@ -1,212 +1,311 @@
-//     const currentParameterTypes = Array.isArray(method.inputs)
-//       ? method.inputs.map(({ type }) => type)
-//       : []
-//     const currentMethodSignature = `${method.name}(${currentParameterTypes.join(
-//       ','
-//     )})`
-//     return currentMethodSignature === methodSignature
-//   })
-
-//   if (!methodJsonDescription) {
-//     throw new Error(`${methodSignature} not found on ABI for ${destination}`)
-//   }
-
-//   return createDirectTransaction(
-//     sender,
-//     destination,
-//     methodJsonDescription,
-//     params
-//   )
-// }
-
-// export function createForwarderTransactionBuilder(
-//   sender,
-//   directTransaction,
-//   web3
-// ) {
-//   const forwardMethod = new web3.eth.Contract(getAbi('aragon/Forwarder'))
-//     .methods['forward']
-
-//   return (forwarderAddress, script) => ({
-//     ...directTransaction, // Options are overwriten by the values below
-//     from: sender,
-//     to: forwarderAddress,
-//     data: forwardMethod(script).encodeABI(),
-//   })
-// }
-
-// export async function getRecommendedGasLimit(
-//   web3,
-//   estimatedGasLimit,
-//   { gasFuzzFactor = DEFAULT_GAS_FUZZ_FACTOR } = {}
-// ) {
-//   const latestBlock = await web3.eth.getBlock('latest')
-//   const latestBlockGasLimit = latestBlock.gasLimit
-
-//   const upperGasLimit = Math.round(
-//     latestBlockGasLimit * PREVIOUS_BLOCK_GAS_LIMIT_FACTOR
-//   )
-//   const bufferedGasLimit = Math.round(estimatedGasLimit * gasFuzzFactor)
-
-//   if (estimatedGasLimit > upperGasLimit) {
-//     // TODO: Consider whether we should throw an error rather than returning with a high gas limit
-//     return estimatedGasLimit
-//   } else if (bufferedGasLimit < upperGasLimit) {
-//     return bufferedGasLimit
-//   } else {
-//     return upperGasLimit
-//   }
-// }
+import ethers from 'ethers'
+
+import TransactionRequest from '../TransactionRequest'
+import App from '../../wrappers/App'
+import { Abi, FunctionFragment } from '../../types'
+
+const DEFAULT_GAS_FUZZ_FACTOR = 1.5
+const PREVIOUS_BLOCK_GAS_LIMIT_FACTOR = 0.95
+
+const erc20ABI = [
+  'function balanceOf(address _who) public view returns (uint256)',
+  'function allowance(address _owner, address _spender) public view returns (uint256)',
+  'function approve(address _spender, uint256 _value) public returns (bool)',
+]
+
+const forwarderAbi = [
+  'function forward(bytes evmCallScript) public',
+  'function canForward(address sender, bytes evmCallScript) public view returns (bool)',
+  'function isForwarder() external pure returns (bool)',
+]
+
+const forwarderFeeAbi = [
+  'function forwardFee() external view returns (address, uint256)',
+]
+
+// Is the given method a full signature, e.g. 'foo(arg1,arg2,...)'
+export const isFullMethodSignature = (methodSignature: string): boolean => {
+  return (
+    Boolean(methodSignature) &&
+    methodSignature.includes('(') &&
+    methodSignature.includes(')')
+  )
+}
```

Comment:
I believe this function fails on this case: `()foo`, so we should also check for the proper placements of the parenthesis.